### PR TITLE
Acrpull kv

### PR DESCRIPTION
**Pull request by wereinse**
## Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

### Validation
- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

### Issues Closed or Referenced

- Closes #<issue number> (this will automatically close the issue when the PR closes)
- References #<issue number> (this references the issue but does not close with PR)

### CODE REVIEW

File: src/modules/acr/main.tf [hunk 1]
```diff
@@ -32,12 +32,26 @@ output "acr" {
...
+}
+resource "azurerm_container_registry_webhook" "webhook" {
+  name                = var.NAME
+  location            = var.LOCATION
+  resource_group_name = var.ACR_RG_NAME
+  registry_name       = azurerm_container_registry.helium-acr.name
+
+  service_uri = "https://${var.NAME}.scm.azurewebsites.net/docker/hook"
+  status      = "enabled"
+  scope       = "helium-${var.NAME}:latest"
```
Comment (bartr) [inline src/modules/acr/main.tf:52]: I think this should be var.Lang

File: src/modules/webapp/main_alerts.tf [hunk 2]
```diff
@@ -101,3 +101,24 @@ resource "azurerm_monitor_metric_alert" "Healthz-webtest-alert" {
     action_group_id = azurerm_monitor_action_group.helium-iac-AG.id
   }
 }
+resource "azurerm_monitor_metric_alert" "Requests-below1-alert" {
+  name                = "${var.NAME}-appinsights-Requests-below1-Alert"
```
Comment (bartr) [inline src/modules/webapp/main_alerts.tf:105]: standardize this name with the rest of the names

File: src/modules/webapp/main_kv.tf [hunk 3]
```diff
@@ -65,7 +65,7 @@ resource "azurerm_key_vault_secret" "cosmoscol" {
 
 resource "azurerm_key_vault_secret" "appinsights" {
   name         = "AppInsightsKey"
-  value        = var.ACR_SP_SECRET
+  value        = azurerm_application_insights.helium.instrumentation_key
```
Comment (bartr) [inline src/modules/webapp/main_kv.tf:68]: this was already fixed - is the branch in sync with master?

File: src/modules/webapp/variables.tf [hunk 4]
```diff
@@ -57,6 +57,11 @@ variable "ACR_SP_SECRET" {
   description = "The ACR Service Principal secret"
 }
 
+variable "TFSTATE_RG_NAME" {
```
Comment (bartr) [inline src/modules/webapp/variables.tf:60]: what does this do?  It's not part of the web app

File: src/modules/webapp/main_kv.tf (hunk 3 above)
Comment (wereinse) [reply src/modules/webapp/main_kv.tf:68]: hadn't seen that you modified - apologies.  did a git pull on master.  I'll fix -

File: src/modules/webapp/variables.tf (hunk 4 above)
Comment (wereinse) [reply src/modules/webapp/variables.tf:60]: it is for the tfstate file that is backed up on azure blob

File: src/modules/acr/variables.tf [hunk 5]
```diff
@@ -13,6 +13,11 @@ variable "ACR_SP_ID" {
   description = "The ACR Service Principal"
 }
 
+variable "ACR_SP_SECRET" {
+  type        = string
+  description = "The ACR Service Secret"
+}
```
Comment (aflinchb) [inline src/modules/acr/variables.tf:19]: Say "Service Principal" to keep consistent with description

File: src/root/main.tf [hunk 6]
```diff
@@ -111,6 +113,11 @@ module "web" {
   WT_SEVERITY       = var.WT_SEVERITY
   WT_THRESHOLD      = var.WT_THRESHOLD
   WT_OPERATOR       = var.WT_OPERATOR
+  WV_FREQUENCY      = var.WT_FREQUENCY
+  WV_WINDOW_SIZE    = var.WT_WINDOW_SIZE
```
Comment (aflinchb) [inline src/root/main.tf:117]: We need to add these values to the readme(s?).

File: src/modules/acr/main.tf [hunk 7]
```diff
@@ -32,12 +32,26 @@ output "acr" {
...
+  }
+}
+resource "azurerm_container_registry_webhook" "webhook" {
+  name                = var.NAME
+  location            = var.LOCATION
+  resource_group_name = var.ACR_RG_NAME
+  registry_name       = azurerm_container_registry.helium-acr.name
+
+  service_uri = "https://${var.LANG}.scm.azurewebsites.net/docker/hook"
+  status      = "enabled"
```
Comment (aflinchb) [inline src/modules/acr/main.tf:51]: Does var.LANG exist? I think it's var.LANGUAGE

File: src/modules/webapp/main_alerts.tf [hunk 8]
```diff
@@ -101,3 +101,24 @@ resource "azurerm_monitor_metric_alert" "Healthz-webtest-alert" {
...
   }
 }
+resource "azurerm_monitor_metric_alert" "Requests-below1-alert" {
+  name                = "${var.NAME}-Requests-below1-Alert"
+  resource_group_name = var.APP_RG_NAME
+  scopes              = [azurerm_application_insights.helium.id]
+  frequency           = var.WV_FREQUENCY
+  window_size         = var.WV_WINDOW_SIZE
+  description         = "Max Requests below 1."
+  severity            = var.WV_SEVERITY
```
Comment (aflinchb) [inline src/modules/webapp/main_alerts.tf:111]: I think removing max from the description is less confusing, so just "Requests below 1"

File: src/modules/acr/main.tf (hunk 7 above)
Comment (bartr) [reply src/modules/acr/main.tf:51]: it should be var.NAME

File: src/modules/webapp/variables.tf (hunk 4 above)
Comment (bartr) [reply src/modules/webapp/variables.tf:60]: Let's leave here for this PR and create a module for TF state

File: src/root/main.tf (hunk 6 above)
Comment (bartr) [reply src/root/main.tf:117]: agree - let's do that in a follow up PR so we can test

**Pull request merged**